### PR TITLE
SE-597 App Upgrade

### LINKED ISSUE #597: Upgrade to latest Evaluator Client Version
Upgrade to the `0.1.1-SNAPSHOT` version of the evaluator Client. Cats and circe version should be upgraded as well.

### PR DESCRIPTION
**Pull request by juanpedromoreno**
Fixes #597 .

Please, @raulraja @dialelo could you review? Thanks!

### CODE REVIEW

File: project/settings.scala [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    pomIncludeRepository := Function.const(false),
+    publishTo := {
+      val nexus = "https://oss.sonatype.org/"
+      if (isSnapshot.value)
+        Some("Snapshots" at nexus + "content/repositories/snapshots")
+      else
+        Some("Releases" at nexus + "service/local/staging/deploy/maven2")
+    },
+    pomExtra :=
+      <developers>
```
Comment (raulraja) [inline project/settings.scala]: Either we add all developers or we remove this list, but it's outdated and misleading. Let's add an issue to take care of it.

File: server/app/org/scalaexercises/exercises/controllers/ApplicationController.scala [hunk 2]
```diff
@@ -79,10 +77,10 @@ class ApplicationController(cache: CacheApi)(
...
       repo ← scalaexercisesRepo
       result ← ops.runFuture map {
-        case Xor.Right((libraries, user, Some(token), progress, _))  ⇒ Ok(views.html.templates.home.index(user = user, libraries = libraries, progress = progress, repo = repo))
-        case Xor.Right((libraries, None, None, progress, authorize)) ⇒ Ok(views.html.templates.home.index(user = None, libraries = libraries, progress = progress, redirectUrl = Option(authorize.url), repo = repo)).withSession("oauth-state" → authorize.state)
-        case Xor.Right((libraries, Some(user), None, _, _))          ⇒ InternalServerError("Session token not found")
-        case Xor.Left(ex) ⇒
+        case Right((libraries, user, Some(token), progress, _))  ⇒ Ok(views.html.templates.home.index(user = user, libraries = libraries, progress = progress, repo = repo))
+        case Right((libraries, None, None, progress, authorize)) ⇒ Ok(views.html.templates.home.index(user = None, libraries = libraries, progress = progress, redirectUrl = Option(authorize.url), repo = repo)).withSession("oauth-state" → authorize.state)
+        case Right((libraries, Some(user), None, _, _))          ⇒ InternalServerError("Session token not found")
```
Comment (raulraja) [inline server/app/org/scalaexercises/exercises/controllers/ApplicationController.scala]: Shouldn't this be `Unauthorized`?

File: server/app/org/scalaexercises/exercises/controllers/ApplicationController.scala [hunk 3]
```diff
@@ -142,10 +140,10 @@ class ApplicationController(cache: CacheApi)(
             contributors = contributors
           )
         ).withSession("oauth-state" → authorize.state)
-      case Xor.Right((Some(l), None, _, _, _, _, _)) ⇒ NotFound("Section not found")
-      case Xor.Right((None, _, _, _, _, _, _))       ⇒ NotFound("Library not found")
-      case Xor.Right((_, _, _, _, _, _, _))          ⇒ InternalServerError("Library and section not found")
-      case Xor.Left(ex) ⇒
+      case Right((Some(l), None, _, _, _, _, _)) ⇒ NotFound("Section not found")
+      case Right((None, _, _, _, _, _, _))       ⇒ NotFound("Library not found")
+      case Right((_, _, _, _, _, _, _))          ⇒ InternalServerError("Library and section not found")
```
Comment (raulraja) [inline server/app/org/scalaexercises/exercises/controllers/ApplicationController.scala]: Shouldn't this be a `NotFound`?

File: server/app/org/scalaexercises/exercises/controllers/AuthenticationModule.scala [hunk 4]
```diff
@@ -45,8 +44,8 @@ trait AuthenticationModule { self: ProdInterpreters ⇒
   def AuthenticatedUser(block: User ⇒ Future[Result])(implicit userOps: UserOps[ExercisesApp], transactor: Transactor[Task]) =
     Secure(AuthenticationAction.async { request ⇒
       userOps.getUserByLogin(request.userId).runFuture flatMap {
-        case Xor.Right(Some(user)) ⇒ block(user)
-        case _                     ⇒ Future.successful(BadRequest("User login not found"))
+        case Right(Some(user)) ⇒ block(user)
```
Comment (raulraja) [inline server/app/org/scalaexercises/exercises/controllers/AuthenticationModule.scala]: We should rename all occurrences of `block` to `thunk` or something more descriptive as block can be misinterpreted as this having blocking semantics in the context of `scala.concurrent.Future`

**Comment by dialelo** [comment]
LGTM :+1:

**Pull request merged**